design/cluster-update.md

```diff
@@ -0,0 +1,209 @@
+We need to be absolutely sure that the user really intended to remove the OSD from a device, as opposed to the device name randomly changing and becoming out of the device filter or list.
+
+What is especially challenging here is that before the initial deployment of OSDs onto a node, which creates the UUIDs for each device, there is no known consistent and user friendly way to specify devices.
+A lot of environments do **not** have labels, IDs, UUIDs, etc. for their devices at first boot and the only way to address them is by device name, such as `sda`.
+This is unfortunate because it is a volatile identifier.
+Some environments do have IDs at first boot and we should consider allowing users to specify devices by those IDs instead of names in the near future.
+That effort is being tracked by [issue 1228](https://github.com/rook/rook/issues/1228).
+
+The main approach that will be taken to solve this issue is to always compare the device UUID from a node's saved OSD config map against the device UUIDs of the current set of device names.
+If the two do not match, then it is not a safe operation to remove the OSD from the device.
+Let's walk through a couple simple scenarios to illustrate this approach:
+
+**NOT SAFE: Device name has changed, but filter has not been updated by the user:**
+* User initially specifies `sda` via device filter or list. Rook configures `sda` and gets an OSD up and running.
+* The node reboots which causes the OSD pod to restart.
+* The filter still specifies `sda`, but the device has changed its name to `sdb`.  The device is now out of the filter.
+* We look at the node's saved OSD config and see that we originally set up `sda` with device UUID `wxyz-1234`.
+* The user's filter still says to use `sda`, so going by the saved config and not what the current devices names are, we know that the old `sda` (device UUID `wxyz-1234`), which is now `sdb` should NOT be removed.
+
+**SAFE: User has updated the filter and the device name has not changed:**
+* User initially specifies `sda` via device filter or list. Rook configures `sda` and gets an OSD up and running.
+* User updates the Cluster CRD to change the device filter or list to now be `sdb`.
+* The OSD pod restarts and when it comes back up it sees that the previously configured `sda` is no longer in the filter.
+* The pod checks the device UUID of `sda` in its saved config and compares that to the device UUID of the current `sda`.
+* The two match, so the pod knows it's a safe (user intended) operation to remove the OSD from `sda`.
+
+### Orchestration
+When the operator receives an event that the Cluster CRD has been updated, it will need to perform some orchestration in order to bring actual state of the cluster in agreement with the desired state.
+For example, if `monCount` were to be updated, the operator would need to add or remove a single monitor at a time, ensuring that quorum is restored before moving onto the next monitor.
```

Comment:
Just as a comment, this is kinda already in the logic of the health check now. If there are less than wanted it spawns more.

Reply:
I clarified the wording a bit.